### PR TITLE
SSCS-10493-Alert-On-Hearing-Event-Failure Initial Setup

### PR DESCRIPTION
**Pull request by MohammedHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10493

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@Slf4j
+@Configuration
+@RequiredArgsConstructor
+@ConditionalOnProperty("flags.tribunals-to-hearings-api.enabled")
+public class TribunalsHearingsEventDeadLetterQueueListener {
+
+    private final TribunalsHearingEventQueueProperties tribunalsHearingEventQueueProperties;
+    private final AppInsightsService appInsightsService;
+    private boolean keepRun = true;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:23]: Optional: Could we rename this to `keepRunning` ?

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        HearingRequest hearingRequest = message.getBody().toObject(HearingRequest.class);
+        Long caseId = Long.valueOf(hearingRequest.getCcdCaseId());
+        return TribunalsDeadLetterMessage.builder()
+            .messageId(message.getMessageId())
+            .body(message.getBody().toString())
+            .caseID(caseId)
+            .build();
+    }
+
+    public void stop() {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:85]: Where does this get called..?

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 1 above)
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:23]: Optional: Can we change this to `keepRunning` ?

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 1 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:23]: Tbh might remove this going to have a catch up with Sean tomorrow and just run it forever was debating it but seen the other team keep it in so thought why not

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 2 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:85]: No where yet was kept in just to have easy access to shut it off if something does need to handle it but most likely the whole thing will be removed after a discussion tomorrow we will see!

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Bean
+    public void processDeadLetterQueue() {
+        try (ServiceBusReceiverClient sessionReceiver = tribunalsHearingsEventProcessorClientDeadLetter()) {
+            while (keepRun) {
+                consumeMessage(sessionReceiver);
+            }
+        } catch (Exception ex) {
+            log.error("Error Processing Queue: " + ex);
+        }
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:33]: Catching "Exception" is a bold thing to do, especially when only logging it.
Consider what else you may need to do. Or even consider the possible exceptions that could be thrown and potentially only catch checked exceptions to allow runtime ones to flow up and be handled higher up.
Where you fully intend on catching "Exception" I'd suggest you make this clear so other dev's better understand

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    } catch (Exception ex) {
+                        log.error("Error processing Dead Letter Queue message with id '{}' - "
+                                      + "abandon the processing", messageId);
+                        receiver.abandon(message);
+                    }
+                }
+            );
+        } catch (Exception ex) {
+            log.info("Failed to consume message with error {}", ex);
+        }
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java:71]: Same comment as above

**Review by nickwest1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 4 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java]: Resolved

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 3 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java]: Resolved

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 2 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java]: Resolved

File: src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java (hunk 1 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java]: Resolved

**Comment by MohammedHMCTS** [comment]
Old Azure Solution has now been changed and updated so PR closed

**Pull request closed without merging**